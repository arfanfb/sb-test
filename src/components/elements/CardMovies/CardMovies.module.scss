.card.movies {
  border-radius: 16px;
  background: white;
  box-shadow: 0px 2px 2px rgba(66, 66, 66, 0.2),
    0px 0px 4px rgba(66, 66, 66, 0.2);
  cursor: pointer;
  width: 255px;

  p {
    margin: 0;
  }

  &.loading {
    .title {
      animation: gradient 1s ease infinite;
      background: linear-gradient(-45deg, #949494, rgba(196, 196, 196, 0));
      background-size: 400% 400%;
      border-radius: 100px;
      display: block;
      height: 14px;
      margin-top: 5px;
    }

    .default {
      animation: gradient 1s ease infinite;
      background: linear-gradient(-45deg, #949494, rgba(196, 196, 196, 0));
      background-size: 400% 400%;
    }

    .year {
      animation: gradient 1s ease infinite;
      background: linear-gradient(-45deg, #949494, rgba(196, 196, 196, 0));
      background-size: 400% 400%;
      border-radius: 100px;
      display: block;
      height: 14px;
      margin-top: 5px;
      width: 100px;
    }
  }

  .default {
    align-items: center;
    background-color: #545454;
    border-top-left-radius: 16px;
    border-top-right-radius: 16px;
    display: flex;
    height: 357px;
    justify-content: center;

    img {
      padding-left: 35px;
    }
  }

  img {
    border-radius: 16px;
    max-height: 357px;
    width: inherit;
  }

  &.info {
    min-height: 429px;

    img {
      border-bottom-left-radius: 0%;
      border-bottom-right-radius: 0%;
    }

    section {
      padding: 0 11px;

      div {
        align-items: center;
        display: flex;
        justify-content: space-between;
        margin-top: 10px;

        svg {
          margin-right: 5px;
        }

        span {
          &:first-child {
            display: flex;
          }
        }
      }
    }
  }

  @media only screen and (max-width: 768px) {
    height: 250px;
    width: 190px;

    img {
      max-height: 250px;
    }

    &.info {
      min-height: 340px;
      height: 340px;

      .icon {
        text-align: right;
      }
    }
  }
}

@keyframes loading-animation {
  0% {
    background-position: -100px;
  }
  40%,
  100% {
    background-position: 350px;
  }
}

@keyframes gradient {
	0% {
		background-position: 0% 50%;
	}
	50% {
		background-position: 100% 50%;
	}
	100% {
		background-position: 0% 50%;
	}
}

