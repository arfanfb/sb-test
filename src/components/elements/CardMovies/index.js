import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import Typography from '../Typography';
import styles from './CardMovies.module.scss';

export default function CardMovies(props) {
  const { 'class-name': classNameProps, 'external-url': externalUrl, title, image, link,
    loading, year } = props;
  const [windows, setWindow] = useState(undefined);

  useEffect(() => {
    setWindow(window);
  },[]);

  const getLink = () => {
    return !externalUrl ? windows.location.origin + '/' + link : link;
  };

  const getTitle = () => {
    if (title) {
      return title;
    }

    return loading ? '' : 'Untitled';
  };

  const renderInfo = () => (
    <section>
      <Typography class-name={styles.title} variant="body-2">
        {getTitle()}</Typography>
      <div>
        {(year && !loading) ? (<Typography variant="overline-1">
          {year}
        </Typography>) : (<span className={styles.year} />)}
      </div>
    </section>
  );

  return (
    <a href={(link && windows) ? getLink() : ''}>
      <div className={[styles.card, styles.movies, classNameProps,
        (title || loading) && styles.info,
        loading && styles.loading].join(' ')}>
        {image ? (<img loading="lazy" src={image}/>) : (
          <span className={styles.default} />
        )}
        {(title || year || loading) && renderInfo()}
      </div>
    </a>
  );
}

CardMovies.defaultProps = {
  'class-name': undefined,
  'external-url': false,
  image: undefined,
  link: undefined,
  loading: false,
  title: null,
  year: null,
};

CardMovies.propTypes = {
  'class-name': PropTypes.oneOfType([PropTypes.object, PropTypes.string]),
  'external-url': PropTypes.bool,
  image: PropTypes.string,
  link: PropTypes.string,
  loading: PropTypes.bool,
  title: PropTypes.string,
  year: PropTypes.string,
};
