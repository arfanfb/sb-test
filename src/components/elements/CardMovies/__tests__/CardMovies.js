import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import CardMovies from '../index';

jest.mock('react', () => ({
  ...require.requireActual('react'),
}));

describe('src/components/elements/CardMovies', () => {
  it('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<CardMovies />);
    expect(tree).toMatchSnapshot();
  });

  it('render with props', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<CardMovies title="Judul Game" year="4" />);
    expect(tree).toMatchSnapshot();
  });

  it('render without title', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<CardMovies year="4" />);
    expect(tree).toMatchSnapshot();
  });

  it('render without joystick', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<CardMovies title="4" />);
    expect(tree).toMatchSnapshot();
  });
});
