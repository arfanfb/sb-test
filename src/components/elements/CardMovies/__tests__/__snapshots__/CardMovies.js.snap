// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/elements/CardMovies render 1`] = `
<React.Fragment>
  <div
    className="   false false"
  >
    <span />
    <Link
      to=""
    />
  </div>
</React.Fragment>
`;

exports[`src/components/elements/CardMovies render with props 1`] = `
<React.Fragment>
  <div
    className="    false"
  >
    <span />
    <Link
      to=""
    >
      <section>
        <Typography
          tag="p"
          variant="body-2"
        >
          Judul Movie
        </Typography>
        <div>
          <Typography
            tag="p"
            variant="overline-1"
          >
            4
          </Typography>
        </div>
      </section>
    </Link>
  </div>
</React.Fragment>
`;

exports[`src/components/elements/CardMovies render without title 1`] = `
<React.Fragment>
  <div
    className="   false false"
  >
    <span />
    <Link
      to=""
    >
      <section>
        <Typography
          tag="p"
          variant="body-2"
        >
          Untitled
        </Typography>
        <div>
          <Typography
            tag="p"
            variant="overline-1"
          >
            4
          </Typography>
        </div>
      </section>
    </Link>
  </div>
</React.Fragment>
`;

exports[`src/components/elements/CardMovies render without year 1`] = `
<React.Fragment>
  <div
    className="    false"
  >
    <span />
    <Link
      to=""
    >
      <section>
        <Typography
          tag="p"
          variant="body-2"
        >
          4
        </Typography>
        <div>
          <span />
        </div>
      </section>
    </Link>
  </div>
</React.Fragment>
`;
